vitrivr-engine-module-features/src/main/kotlin/org/vitrivr/engine/base/features/external/common/ExternalWithFloatVectorDescriptorAnalyser.kt

```diff
@@ -38,9 +38,18 @@ abstract class ExternalWithFloatVectorDescriptorAnalyser<C : ContentElement<*>>
      * @param requestBody The body of the API request.
      * @return A list of floats representing the API response.
      */
-    private fun executeApiRequest(url: String, requestBody: String): List<Value.Float> {
+    private fun executeApiRequest(content: ContentElement<*>, url: String): List<Value.Float> {
+
+        val base64 = when (content) {
+            is TextContent -> Base64.getEncoder().encodeToString(content.content.toByteArray(StandardCharsets.UTF_8))
+            is ImageContent -> content.content.toDataURL()
+            else -> throw IllegalArgumentException("Unsupported content type")
+        }
+
+        val requestBody = URLEncoder.encode("data:text/plain;charset=utf-8,$base64", StandardCharsets.UTF_8.toString())
```

Comment:
Yes! 

This is copied from @rahelarnold98 implementation. I'm pretty sure it works but I agree (since we used the service), but I agree that it is not the proper way of doing it.

Reply:
I see, so it's something that would need to be verified on the other side as well.

Reply:
Indeed. I'll have to check with @rahelarnold98